react/OrderShipping.tsx

```diff
@@ -1,31 +1,202 @@
-import React, { createContext, ReactNode, useContext } from 'react'
-
+import React, {
+  createContext,
+  ReactNode,
+  useContext,
+  useCallback,
+  useEffect,
+  useRef,
+} from 'react'
+import { compose, graphql } from 'react-apollo'
+import { useOrderQueue } from 'vtex.order-manager/OrderQueue'
 import { useOrderForm } from 'vtex.order-manager/OrderForm'
 
+import {
+  estimateShipping as EstimateShipping,
+  selectDeliveryOption as SelectDeliveryOption,
+} from 'vtex.checkout-resources/Mutations'
+
 interface Context {
   countries: string[]
+  selectedAddress: CheckoutAddress
+  insertAddress: any
+  deliveryOptions: DeliveryOption[]
+  selectDeliveryOption: any
 }
 
 interface OrderShippingProps {
   children: ReactNode
+  EstimateShipping: any
+  SelectDeliveryOption: any
 }
 
 const OrderShippingContext = createContext<Context | undefined>(undefined)
 
-export const OrderShippingProvider = ({ children }: OrderShippingProps) => {
-  const {
-    orderForm: {
-      shipping: { countries },
-    },
-  } = useOrderForm()
-
-  return (
-    <OrderShippingContext.Provider value={{ countries }}>
-      {children}
-    </OrderShippingContext.Provider>
-  )
+const shippingId = 'shipping'
+
+const getNewDeliveryOptions = (
+  deliveryOptions: DeliveryOption[],
+  selectedDeliveryOption: string
+) => {
+  return deliveryOptions.map(option => {
+    if (option.isSelected) {
+      option.isSelected = false
```

Comment:
Does it make sense?

Reply:
This function aims at changing the selected delivery option to the one that has the given `deliveryOptionId`. So, I thought it would be better to rename it to `changeSelectedDeliveryId`. But I don't know if that's what you're talking about.